db/migrate/20181212165840_create_internal_notes.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateInternalNotes < ActiveRecord::Migration[5.2]
+  def change
+    create_table :internal_notes do |t|
+      t.text :body, null: false
+      t.references :document, foreign_key: { on_delete: :cascade }, null: false
+      t.references :user, foreign_key: { on_delete: :restrict }
```

Comment:
I think this should probably be a nullify rather than a restrict - I'm guessing though you've done it in a way that's consistent with retire and remove.

So if this is consistent I'd suggest it's good and we'll try resolve it and them at same time, but if it's different it probably should be nullify as I guess we want to allow users to be removed from the system if they so chose.

Reply:
Yep that makes sense, i followed the pattern as done with retire and remove.